tests/store/artifact/test_databricks_artifact_repo.py

```diff
@@ -591,6 +556,226 @@ def test_paginated_list_artifacts(self, databricks_artifact_repo):
+                    mock.call(GetCredentialsForRead(run_id=MOCK_RUN_ID, path=paths_chunk_3)),
+                ]
+            )
+
+    def test_get_write_credential_infos_handles_pagination(self, databricks_artifact_repo):
+        """
+        Verifies that the `_get_write_credential_infos` method, which is used to resolve write
+        access credentials for a collection of artifacts, handles paginated responses properly,
+        issuing incremental requests until all pages have been consumed
+        """
+        credential_infos_mock_1 = [
+            ArtifactCredentialInfo(
+                signed_uri="http://mock_url_1", type=ArtifactCredentialType.AWS_PRESIGNED_URL
+            ),
+            ArtifactCredentialInfo(
+                signed_uri="http://mock_url_2", type=ArtifactCredentialType.AWS_PRESIGNED_URL
+            ),
+        ]
+        credential_infos_mock_2 = [
+            ArtifactCredentialInfo(
+                signed_uri="http://mock_url_3", type=ArtifactCredentialType.AWS_PRESIGNED_URL
+            )
+        ]
+        credential_infos_mock_3 = []
+
+        with mock.patch(
+            DATABRICKS_ARTIFACT_REPOSITORY_PACKAGE + ".message_to_json"
+        ) as message_mock, mock.patch(
+            DATABRICKS_ARTIFACT_REPOSITORY + "._call_endpoint"
+        ) as call_endpoint_mock:
+            get_credentials_for_write_responses = [
+                GetCredentialsForWrite.Response(
+                    credential_infos=credential_infos_mock_1, next_page_token="2"
+                ),
+                GetCredentialsForWrite.Response(
+                    credential_infos=credential_infos_mock_2, next_page_token="3"
+                ),
+                GetCredentialsForWrite.Response(credential_infos=credential_infos_mock_3,),
+            ]
+            call_endpoint_mock.side_effect = get_credentials_for_write_responses
+            write_credential_infos = databricks_artifact_repo._get_write_credential_infos(
+                MOCK_RUN_ID, ["testpath"],
+            )
+            assert write_credential_infos == credential_infos_mock_1 + credential_infos_mock_2
+            message_mock.assert_has_calls(
+                [
+                    mock.call(GetCredentialsForWrite(run_id=MOCK_RUN_ID, path=["testpath"])),
+                    mock.call(
+                        GetCredentialsForWrite(
+                            run_id=MOCK_RUN_ID, path=["testpath"], page_token="2"
+                        )
+                    ),
+                    mock.call(
+                        GetCredentialsForWrite(
+                            run_id=MOCK_RUN_ID, path=["testpath"], page_token="3"
+                        )
+                    ),
+                ]
+            )
+            assert call_endpoint_mock.call_count == 3
+
+    def test_get_write_credential_infos_respects_max_request_size(self, databricks_artifact_repo):
```

Comment:
As discussed in https://github.com/mlflow/mlflow/pull/4402/files#r641201273, these are the other 2 of 4 new test cases